### DISCUSSION TITLE
idea: add a toggle so that pyright ignores narrowed mypy type:ignore annotations

### DISCUSSION THREAD

**Discussion by danielbprice** [microsoft/pyright#8863] [Ideas]
I recently reviewed the open issues and forum discussions about the issues over the interpretation of `type: ignore` comments.  I maintain a source base where Pyright is the primary type checker but at times we also run mypy to see if it finds anything else.

A typical example from our source code would be:
```python
# The azure code post-processes this attribute into the exception, so pyright doesn't know about it.
error_code = e.error_code  # pyright: ignore[reportAttributeAccessIssue]
```

mypy also finds and surfaces this non-issue.  I could simply add `type: ignore` to silence both checkers but as a long time user of lint and other checking tools, I feel strongly that the best practice is to silence just the specific error; this also helps the code be more self-documenting.

The issues here stem from mypy's non-standard extension `# type: ignore[specific-issue]`.  This is discussed here: https://github.com/python/mypy/issues/12358 and elsewhere.  I can't add `# type: ignore[something-specific-to-mypy]` without pyright treating this as `# type: ignore` (suppress all issues).  mypy is (dubiously) extending the spec, and pyright is trying to adhere to the spec as written.

After reviewing the differences of opinion between the various parties, I wondered if Pyright could bend slightly here, and add a small configuration knob to say: if I see a narrowed mypy `# type: ignore[something]` statement, instead of treating it as a valid PEP484 comment, _treat it as invalid_.  And drop it on the floor.  Then it would be possible to have:

```
error_code = e.error_code  # type: ignore[attr-defined] # pyright: ignore[reportAttributeAccessIssue]
                               ^ pyright sees this as         ^ mypy ignores this
                                 mypy PEP-non-compliant
                                 and ignores
```

A user who opts to stick `type: ignore` here, without the mypy-specific narrowing, would have things work as before.  Something like `ignore_mypy_extended_type_ignores = true` (uhh, naming is hard).

I realize that this request is putting the onus on Pyright to solve a problem caused by mypy, but I didn't see this suggestion in the previous discussions, and I thought it might provide an escape hatch for everyone who is affected until such time as mypy implements `mypy:` comments or a new PEP with further clarifications is produced.

Thanks.  One way or another, I really appreciate this tool!

**Comment by erictraut** [comment]
Pyright provides a setting `enableTypeIgnoreComments`. If you set this to false, it will treat all `# type: ignore` comments as no-ops. This doesn't affect `# pyright: ignore` statements. I think that's effectively what you're asking for here.

**Reply by danielbprice** [reply]
Thank you-- I think that will do the trick!
